runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -1,12 +1,196 @@
+
+        long subscriptionTimestamp = getSubscriptionTimestamp(corfuStore, namespace, clientListener);
+        corfuStore.getRuntime().getTableRegistry().getStreamingManager().subscribeLogReplicationListener(
+                clientListener, namespace, streamTag, tablesOfInterest, subscriptionTimestamp, bufferSize);
+        log.info("Client subscription at timestamp {} successful.", subscriptionTimestamp);
+    }
+
+
+    private static long getSubscriptionTimestamp(CorfuStore corfuStore, String namespace,
+                                                 LogReplicationListener clientListener) {
+        Optional<Counter> mvoTrimCounter = MicroMeterUtils.counter("logreplication.subscribe.trim.count");
+        Optional<Counter> conflictCounter = MicroMeterUtils.counter("logreplication.subscribe.conflict.count");
+        Optional<Timer.Sample> subscribeTimer = MicroMeterUtils.startTimer();
+
+        Table<LogReplicationSession, ReplicationStatus, Message> replicationStatusTable =
+            openReplicationStatusTable(corfuStore);
+
+        try {
+            return IRetry.build(IntervalRetry.class, () -> {
+                try (TxnContext txnContext = corfuStore.txn(namespace)) {
+                    // The transaction is started in the client's namespace and the Replication Status table resides in the
+                    // system namespace.  Corfu Store does not validate the cross-namespace access as long as there are no
+                    // writes on the table in the different namespace.  This hack is required here as we want client full
+                    // sync to happen in the same transaction which checks the status of a snapshot sync so that there is no
+                    // window between the check and full sync.
+                    List<CorfuStoreEntry<LogReplicationSession, ReplicationStatus, Message>> entries =
+                        txnContext.executeQuery(replicationStatusTable,
+                                record -> record.getKey().getSubscriber().getModel()
+                                    .equals(LogReplication.ReplicationModel.LOGICAL_GROUPS));
+
+                    CorfuStoreEntry<LogReplicationSession, ReplicationStatus, Message> entry = null;
+
+                    // It is possible that there is no entry for the Logical Group Model in the status table in
+                    // the following scenarios:
+                    // 1. Request to subscribe is received before the Log Replication JVM or pod starts and
+                    // initializes the table
+                    // 2. Topology change which removes this cluster from the status table.
+                    // We cannot differentiate between the two cases here so continue with the right
+                    // behavior for the 1st case, i.e., subscribe and wait for the initial Snapshot Sync to get
+                    // triggered.  If we are here because of the second case, the listener will not get any updates
```

Comment:
The no-op statement is not true, right? imagine Routing_queue model is running any update to that session will cause this to trigger, which in turn not finding LG will cause performFullSync to run on each session update, couldn't this lead to an issue? (performance/resources)

Reply:
Routing queue model could interfere, cause we're subscribing to replication status table (any update coming to that table) right?.. we are not subscribed to a specific key in the table but to any update coming into the metadata table.

Reply:
I thought that was already filtered at L72?

Reply:
We will get updates on the metadata table but they also get filtered based on the replication model and client name.  Updates from the routing queue model will not be processed.  
https://github.com/CorfuDB/CorfuDB/blob/lr_listener/runtime/src/main/java/org/corfudb/runtime/LogReplicationListener.java#L124

Reply:
Correct, I put the comment when client name was not part of the PR, and all logical_group sessions would filter down. But with that this issue is no longer there.